tests/integration/ha/test_backups.py

```diff
@@ -114,44 +119,201 @@ async def c_writes_runner(ops_test: OpsTest, c_writes: ContinuousWrites):
     """Starts continuous write operations and clears writes at the end of the test."""
     await c_writes.start()
     yield
+
+    reachable_ip = random.choice(await get_reachable_unit_ips(ops_test))
+    await http_request(ops_test, "GET", f"https://{reachable_ip}:9200/_cat/nodes", json_resp=False)
+    await http_request(
+        ops_test, "GET", f"https://{reachable_ip}:9200/_cat/shards", json_resp=False
+    )
+
     await c_writes.clear()
     logger.info("\n\n\n\nThe writes have been cleared.\n\n\n\n")
 
 
+@pytest.fixture(scope="session")
+def microceph():
+    """Starts microceph radosgw."""
+    import subprocess
+
+    if "microceph" not in subprocess.check_output(["sudo", "snap", "list"]).decode():
+        import os
+
+        import requests
+
+        uceph = "/tmp/microceph.sh"
+
+        with open(uceph, "w") as f:
+            resp = requests.get(
+                "https://raw.githubusercontent.com/canonical/microceph-action/main/microceph.sh"
+            )
+            f.write(resp.content.decode())
+
+        os.chmod(uceph, 0o755)
+        subprocess.check_output(
+            [
+                "sudo",
+                uceph,
+                "-c",
+                "latest/edge",
+                "-d",
+                "/dev/sdi",
+                "-a",
+                "accesskey",
+                "-s",
+                "secretkey",
+                "-b",
+                "data-charms-testing",
+                "-z",
+                "5G",
+            ]
+        )
+    # Now, return the configuration
+    ip = subprocess.check_output(["hostname", "-I"]).decode().split()[0]
+    return {"url": f"http://{ip}", "access-key": "accesskey", "secret-key": "secretkey"}
+
+
+async def _wait_backup_finish(ops_test, leader_id):
```

Comment:
tests should start with `test_` they should also be decorated with `@pytest.mark.abort_on_fail` - please refer to the rest of the integration tests for reference.

Reply:
Erh, not sure I get it, that is an aux. method that pretty much every test in `test_backups.py` calls at some point... It is not intended to be a test itself.
I will move it to `ha/helpers.py`